api/base/serializers.py

```diff
@@ -88,6 +89,26 @@ def to_internal_value(self, data):
         return super(JSONAPIListField, self).to_internal_value(data)
 
 
+class AuthorizedCharField(ser.CharField):
```

Comment:
To address (2), we could do something like `AuthorizedField(CharField(), ....)`, but that feels a little over-engineered for  something we will only use once. As suggested in the previous comment, I think a `SerializerMethodField` is the way to go.